src/_pytest/assertion/rewrite.py

```diff
@@ -217,8 +194,9 @@ def _should_rewrite(self, name, fn_pypath, state):
 
         # modules not passed explicitly on the command line are only
         # rewritten if they match the naming convention for test files
+        fn_path = PurePath(fn)
```

Comment:
Could instantiate the `PurePath` in the beginning already, and use it for the basename check already.

Reply:
yeah, seems not worth it -- I only converted to PurePath here because the function below requires it

(personally I don't like pathlib, but that's just me)